--- a/feeds/packages/utils/dockerd/files/dockerd.init	2026-09-19 08:16:50.962095380 +0800
+++ b/feeds/packages/utils/dockerd/files/dockerd.init	2026-09-19 08:16:50.970246907 +0800
@@ -190,6 +190,8 @@
 	config_get https_proxy proxies https_proxy "${https_proxy}"
 	config_get no_proxy proxies no_proxy "${no_proxy}"
 	config_get storage_driver globals storage_driver ""
+	config_get_bool no_new_privileges globals no_new_privileges "0"
+	config_get_bool live_restore globals live_restore "0"
 
 	. /usr/share/libubox/jshn.sh
 	json_init
@@ -220,6 +222,8 @@
 		json_close_object
 	fi
 	[ -z "${storage_driver}" ] || json_add_string "storage-driver" "${storage_driver}"
+	[ "${no_new_privileges}" != "1" ] || json_add_boolean "no-new-privileges" 1
+	[ "${live_restore}" != "1" ] || json_add_boolean "live-restore" 1
 	json_dump > "${DOCKERD_CONF}"
 
 	[ "${iptables}" -eq "1" ] && config_foreach iptables_add_blocking_rule firewall
